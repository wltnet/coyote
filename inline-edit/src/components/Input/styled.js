import styled from "styled-components";

export const Label = styled.label`
`;

export const InputField = styled.input`
  font-size: 1rem;
`;
