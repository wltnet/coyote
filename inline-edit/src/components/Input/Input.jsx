import React from "react";
import { Label, InputField } from "./styled";

const Input = ({ value, inputField, handleBlur, handleChange, handleKeyDown }) => (
  <Label>
    <InputField
      type="text"
      name={inputField.name}
      value={value}
      onChange={handleChange}
      onBlur={handleBlur}
      onKeyDown={handleKeyDown}
      autoFocus
    />
  </Label>
);
export default Input;
