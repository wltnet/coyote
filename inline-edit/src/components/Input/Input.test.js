import React from "react";
import { shallow } from "enzyme";
import Input from "./Input";

jest.mock("./styled", () => ({
  Label: "Label",
  InputField: "InputField",
}));

const props = {
  value: "Hello World",
  inputField: {
    label: "helloInput",
    name: "helloInput",
  },
  handleBlur: jest.fn(),
  handleChange: jest.fn(),
  handleKeyDown: jest.fn(),
}

describe("<Input />", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<Input {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleBlur when on blur", () => {
    const wrapper = shallow(<Input {...props} />);
    wrapper.find("InputField").simulate("blur");
    expect(props.handleBlur).toHaveBeenCalled();
  });

  it("should call handleChange when input text has changed", () => {
    const wrapper = shallow(<Input {...props} />);
    wrapper.find("InputField").simulate("change");
    expect(props.handleChange).toHaveBeenCalled();
  });

  it("should call handleKeyDown when key has press", () => {
    const wrapper = shallow(<Input {...props} />);
    wrapper.find("InputField").simulate("keyDown", {key: 'Enter'});
    expect(props.handleKeyDown).toHaveBeenCalled();
  });

});