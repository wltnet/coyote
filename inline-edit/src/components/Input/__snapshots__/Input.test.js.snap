// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match the snapshot 1`] = `
<Label>
  <InputField
    autoFocus={true}
    name="helloInput"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onKeyDown={[MockFunction]}
    type="text"
    value="Hello World"
  />
</Label>
`;
